lib/chef/resource_reporter.rb

```diff
@@ -271,6 +276,12 @@ def prepare_run_data
       run_data["data"] = {}
       run_data["start_time"] = start_time.to_s
       run_data["end_time"] = end_time.to_s
+      if @expanded_run_list
+        run_data["expanded_run_list"] = @expanded_run_list.parse_into_json
+      else 
+        run_data["expanded_run_list"] = {}
+      end
```

Comment:
I think if you initialize `@expanded_run_list = {}` and then just `run_data["expanded_run_list"] = @expanded_run_list` that it'll encode to JSON correctly.